core/sail/lucene/src/main/java/org/eclipse/rdf4j/sail/lucene/LuceneIndex.java

```diff
@@ -753,11 +751,15 @@ public DocumentScore apply(ScoreDoc doc) {
 		SpatialStrategy strategy = getSpatialStrategyMapper().apply(geoField);
 		final Shape boundingCircle = strategy.getSpatialContext().getShapeFactory().circle(p, degs);
 		Query q = strategy.makeQuery(new SpatialArgs(SpatialOperation.Intersects, boundingCircle));
+
+		// use LatLonPoint for distance query after indexing it with the same data structure
+
+		q = LatLonPoint.newDistanceQuery(geoField, p.getY(), p.getX(), distance);
```

Comment:
This brings up a point: are these changes compatible with existing indexes? What would happen if you ran this code on a Lucene index that was generated with an older version of RDF4J?

Reply:
@jeenbroekstra  In this scenario it would not work and basically it is better to re-index things in Lucene because in the current implementation it was failing to index all other geometries - check the issue #2489 . Adding to this, that I tried to index some polygons using the current indexing algorithm after doing some modifications and it took a very long time while using this new algorithm it is much faster. That is to say that this modification is worth it and if you would like to have a turnaround to still support old index at search time, one have to figure out the shape of the index and then choose to search using the old or the new algorithm which is something not so practical.

Reply:
Oh I'm not disputing the usefulness of this fix, the reason I asked is that I wanted to know if we will need to put anything in the release notes about updating existing indexes. Which, apparently, we do, so I'll make a note of that.